--- config.tmpl.orig	2014-11-16 22:49:02.000000000 +0100
+++ config.tmpl	2020-03-08 02:48:48.159903852 +0100
@@ -40,8 +40,7 @@
 <div class="container padding" style="width: 600px;">
     <table class="infoTable">
         <tr>
-            <td><a href="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=JA8M7VDY89SQ4" rel="noreferrer" onclick="window.open('${sickbeard.ANON_REDIRECT}' + this.href, '_blank'); return false;"><img src="$sbRoot/images/paypal/btn_donateCC_LG.gif" width="147" height="47" alt="[donate]" /></a></td>
-            <td>Sick Beard is free, but you can contribute by giving a <b><a href="https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&amp;hosted_button_id=JA8M7VDY89SQ4" rel="noreferrer" onclick="window.open('${sickbeard.ANON_REDIRECT}' + this.href, '_blank'); return false;">donation</a></b>.</td>
+            <td>&nbsp;</td>
         </tr>
     </table>
 </div>
